PatientMonitoringSystem/PatientMonitoringSystem/Database/DatabaseHandler.cs

```diff
@@ -0,0 +1,65 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using System.Data;
+using System.Data.SqlClient;
+using System.Configuration;
+
+namespace PatientMonitoringSystem.Database
+{
+	public class DatabaseHandler
+	{
+
+		private static DatabaseHandler db_instance = null;
+		private static SqlConnection connection;
+		private static readonly object _lock = new object();
+		private static String connection_str;
+		private DatabaseHandler()
+		{
+			string connection_str = @"Data Source=(LocalDB)\MSSQLLocalDB;AttachDbFilename=|DataDirectory|\Database\Database1.mdf;Integrated Security=True";
+			connection = new SqlConnection(connection_str);
+		}
+
+		public static DatabaseHandler Instance
+		{
+			get
+			{
+				lock (_lock)
```

Comment:
@harrytpstudent Good use of locking for thread safety.

Reply:
@harrytpstudent Another way of avoiding race conditions when implementing singletons is to use a static constructor, e.g. `static DatabaseHandler() { /* initialisation code */ }`, or static field initialiser, e.g. `private readonly static DatabaseHandler db_instance = /* initialisation code */;`, which is thread-safe and guaranteed to only be called once. Then you could mark your `db_instance` field as `readonly`.